### PR TITLE
[LoDash/fp] Add LodashConvertible interface to allow calling .convert({}) on each lodash/fp function

### LINKED ISSUE #27194: Lodash/fp convert method not typed
The `convert` function that should be available on each `lodash/fp` exported member is not typed correctly for `mapValues` (the only one I checked but probably others too)

I should be able to do the following:
```js
import { mapValues } from 'lodash/fp'

const mapValuesUncapped = mapValues.convert({ cap: false })
// I get the error => `[ts] Property 'convert' does not exist on type 'LodashMapValues'.`
```

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [ ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @bczengel @chrootsu @stepancar @aj-r @Ailrun @e-cloud @thorn0 @jtmthf @DomiR

### PR DESCRIPTION
**Pull request by toniopelo**
Fixes: #27194

Please fill in this template.
Here a link to the [lodash/fp guide](https://github.com/lodash/lodash/wiki/FP-Guide#convert) talking about the `.convert({})` method I just typed in this PR.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

----

This is a revival of #27224 with the latest review changes addressed.
`convert()` now returns a `(...args: any[]) => any` typed function as I didn't figure out any simple workaround to this.
I guess that in the `processDefinition` function of `generate-fp` we might create specific `convert()` methods signature based on the information we got locally.
My limited knowledge and free time is limiting me here, if you have any clue don't hesitate, feel free to improve this PR.

### CODE REVIEW

File: types/lodash/scripts/generate-fp.ts [hunk 1]
```diff
@@ -113,6 +117,10 @@ async function main() {
         "",
         "declare const _: _.LoDashFp;",
         "declare namespace _ {",
+        // Add LodashConvertible to allow `.convert` method on each lodash/fp function
+        "    interface LodashConvertible {",
+        "        convert: (options: lodash.ConvertOptions) => (...args: any[]) => any;",
```
Comment (aj-r) [inline types/lodash/scripts/generate-fp.ts:122]: nit: can you define this with non-arrow syntax? e.g.

```
convert(options: lodash.ConvertOptions):  (...args: any[]) => any;
```

File: types/lodash/tsconfig.json [hunk 2]
```diff
@@ -19,6 +19,391 @@
     },
     "files": [
         "index.d.ts",
-        "lodash-tests.ts"
+        "lodash-tests.ts",
+        "fp/add.d.ts",
```
Comment (aj-r) [inline types/lodash/tsconfig.json:23]: These file references were recently removed. I'm not sure if they're actually needed, would like to hear from the TypeScript team on this. FYI there is a PR in progress to stop auto-generating these.

**Review by aj-r — CHANGES_REQUESTED** [review]

File: types/lodash/tsconfig.json (hunk 2 above)
Comment (toniopelo) [reply types/lodash/tsconfig.json:23]: I'm going to leave only the index.d.ts file here as it seems to be breaking the CI process too 😌

**Comment by toniopelo** [comment]
@aj-r Updated

**Review by aj-r — APPROVED** [review]

**Review by XVincentX — APPROVED** [review]
Great, this is exactly what I also was looking for.

**Comment by vomvoru** [comment]
LodashConvertible is not available in typescript 3.1 or higher 😢

**Comment by amcasey** [comment]
@vomvoru can you please provide more details about what you tried and how the outcome differed from your expectations?

**Comment by aj-r** [comment]
I think we need to make the same change to the types/lodash/ts3.1 folder as
well.

**Comment by amcasey** [comment]
@aj-r Thanks for clarifying!  Perhaps either you or @vomvoru would like to create an issue (on this repo) and tag the owners of the ts3.1 types (assuming @toniopelo doesn't want to just port them immediately).

**Comment by toniopelo** [comment]
I didn't catch the typescript 3.1 folder, thanks for the heads up!
I created another PR, basically the same but in the `ts3.1` folder

cc @aj-r @amcasey @vomvoru

**Pull request merged**